core/src/test/scala/unit/kafka/server/KafkaMetricReporterClusterIdTest.scala

```diff
@@ -52,8 +53,21 @@ object KafkaMetricReporterClusterIdTest {
     override def onUpdate(clusterMetadata: ClusterResource) {
       MockBrokerMetricsReporter.CLUSTER_META.set(clusterMetadata)
     }
-  }
 
+    override def configure(configs: java.util.Map[String, _]): Unit = {
+      // Check that the configuration passed to the MetricsReporter includes the broker id as an Integer.
+      // This is a regression test for KAFKA-4756.
```

Comment:
The broker config in line 66 actually uses a manually configured broker id. To test the fix, perhaps we should set brokerId to -1 to trigger the auto id generation logic?